cou/utils/nova_compute.py

```diff
@@ -60,3 +64,17 @@ async def get_instance_count(unit: str, model: COUModel) -> int:
         f"No valid instance count value found in the result of {action_name} action "
         f"running on '{unit}': {action.results}"
     )
+
+
+async def get_instance_count_to_upgrade(unit: ApplicationUnit, model: COUModel) -> None:
+    unit_instance_count = await get_instance_count(unit, model)
+    if unit_instance_count != 0:
+        model.run_action(unit_name=unit.name, action_name="enable", raise_on_failure=True)
```

Comment:
Maybe we will need to try catch this and if that failed, send `SIGINT` or `SIGTERM`. But maybe it's not needed, and it's safe to continue even if it failed.